frontend/src/pages/admin/AdminPage.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import useAdminStore from '../../store/admin.store';
+import style from './admin.module.css';
+import CountDisplay from '../../components/adminDetails/countDisplay/CountDisplay';
+import JobApproval from '../../components/adminDetails/jobApproval/JobApproval';
+
+function AdminPage() {
+  const jobsRequest = useAdminStore((state) => state.jobsRequest);
+  const getJobRequest = useAdminStore((state) => state.getJobRequest);
+  const usersCount = useAdminStore((state) => state.usersCount);
+  const getUserCount = useAdminStore((state) => state.getUserCount);
+  const companiesCount = useAdminStore((state) => state.companiesCount);
+  const getComaniesCount = useAdminStore((state) => state.getComaniesCount);
+  const jobsCount = useAdminStore((state) => state.jobsCount);
+  const getJobsCount = useAdminStore((state) => state.getJobsCount);
+  const openJobsCount = useAdminStore((state) => state.openJobsCount);
+  const getOpenJobsCount = useAdminStore((state) => state.getOpenJobsCount);
+
+  useEffect(() => {
+    getJobRequest();
```

Comment:
You can incorporate this logic into a separate hook named ﻿useGetAdminData within the admin folder. In this hook, you should call all the necessary methods and return their results to ﻿AdminPage. This might not seem significant now, but this decoupling of fetching and rendering processes can make your code more testable in the future and of course easier to read :D

Reply:
The benefits of using custom hooks includes keeping the DRY (Do Not Repeat Yourself) convention, which is particularly useful when you need to use the same logic across multiple modules. Custom hooks assist in separating UI components from functional elements, thereby enhancing code readability. Moreover, they improve testability, among other advantages. Let me illustrate these concepts with the ﻿`useAdminData` hook how we can incorporate it into the current project architecture.

1. create a folder `hooks` in the `src` directory
2. lets create a new file `useAdminData`
3. extract the logic from AdminPage and move it into the file
```
import { useEffect } from 'react';
import useAdminStore from '../store/admin.store';

export const useAdminData = () => {
  const jobsRequest = useAdminStore((state) => state.jobsRequest);
  const getJobRequest = useAdminStore((state) => state.getJobRequest);
  const usersCount = useAdminStore((state) => state.usersCount);
  const getUserCount = useAdminStore((state) => state.getUserCount);
  const companiesCount = useAdminStore((state) => state.companiesCount);
  const getComaniesCount = useAdminStore((state) => state.getComaniesCount);
  const jobsCount = useAdminStore((state) => state.jobsCount);
  const getJobsCount = useAdminStore((state) => state.getJobsCount);
  const openJobsCount = useAdminStore((state) => state.openJobsCount);
  const getOpenJobsCount = useAdminStore((state) => state.getOpenJobsCount);

  useEffect(() => {
    getJobRequest();
    getUserCount();
    getComaniesCount();
    getJobsCount();
    getOpenJobsCount();
  }, []);


  return { usersCount, companiesCount, jobsCount, jobsRequest, openJobsCount }
}
```
4. import useAdminData hook back into Admin page and use the data from it.
`const { usersCount, companiesCount, jobsCount, jobsRequest, openJobsCount } = useAdminData()
 `

Now whenever you need those data to be fetched or accessed you can simple re-use this hook 
PS: I am still not sure if we really need the keep the data in zustand storage

Reply:
Great Idea , Thank you so much for explaining :heart: . I will make the changes